# 导入库

In [1]:
import pandas as pd;
import pyecharts.options as opts
from pyecharts.charts import Line

# 加载数据

In [2]:
df1 = pd.read_excel(r'D:/桌面/BIM数据分析/1.xlsx')
food = pd.read_excel(r'D:/桌面/BIM数据分析/食物.xlsx')
sport = pd.read_excel(r'D:/桌面/BIM数据分析/运动.xlsx')

# 原数据展示

In [3]:
df1.head()

日期  体重（kg）         早餐           午餐            晚餐           运动
0 2023-06-01   70.20   鸡蛋三明治、牛奶   煮鸡胸肉、青菜、米饭  烤三文鱼、西兰花、烤土豆     跑步（30分钟）
1 2023-06-02   70.10  燕麦片、酸奶、水果  凉拌黄瓜、炒牛肉、米饭    煮虾、炒菜花、烤红薯     游泳（45分钟）
2 2023-06-03   70.30   煎蛋、面包、果汁     番茄炒蛋、炒米粉   煮牛肉、炒蔬菜、烤南瓜     瑜伽（60分钟）
3 2023-06-04   70.00   炒饭、豆浆、水果   煮鸡蛋、炒青菜、米饭  烤鸡胸肉、烤蔬菜、烤豆腐  健身房训练（30分钟）
4 2023-06-05   69.89   面条、煮蛋、果汁  炒鸡胸肉、炒青菜、米饭   煮鱼、凉拌黄瓜、烤番薯   骑自行车（45分钟）

In [4]:
food.head()

食物  摄入卡路里（大卡/100克）
0   燕麦片             389
1    酸奶              63
2    水果              60
3  煮鸡胸肉             165
4   炒杂菜              40

In [5]:
sport.head()

运动  每分钟消耗卡路里（大卡/分钟）
0     跑步             10.0
1     游泳              9.2
2  健身房训练              8.0
3     瑜伽              3.0
4   骑自行车              8.5

# 数据清洗

* 检查并去除null值

In [6]:
df1.shape

(30, 6)

In [7]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 30 entries, 0 to 29
Data columns (total 6 columns):
 #   Column  Non-Null Count  Dtype         
---  ------  --------------  -----         
 0   日期      30 non-null     datetime64[ns]
 1   体重（kg）  30 non-null     float64       
 2   早餐      30 non-null     object        
 3   午餐      30 non-null     object        
 4   晚餐      30 non-null     object        
 5   运动      30 non-null     object        
dtypes: datetime64[ns](1), float64(1), object(4)
memory usage: 1.5+ KB


* 标准化列标签

In [8]:
df1.columns = ['date','weight(kg)','breakfast','lunch','dinner','sport']
df1.head()

date  weight(kg)  breakfast        lunch        dinner        sport
0 2023-06-01       70.20   鸡蛋三明治、牛奶   煮鸡胸肉、青菜、米饭  烤三文鱼、西兰花、烤土豆     跑步（30分钟）
1 2023-06-02       70.10  燕麦片、酸奶、水果  凉拌黄瓜、炒牛肉、米饭    煮虾、炒菜花、烤红薯     游泳（45分钟）
2 2023-06-03       70.30   煎蛋、面包、果汁     番茄炒蛋、炒米粉   煮牛肉、炒蔬菜、烤南瓜     瑜伽（60分钟）
3 2023-06-04       70.00   炒饭、豆浆、水果   煮鸡蛋、炒青菜、米饭  烤鸡胸肉、烤蔬菜、烤豆腐  健身房训练（30分钟）
4 2023-06-05       69.89   面条、煮蛋、果汁  炒鸡胸肉、炒青菜、米饭   煮鱼、凉拌黄瓜、烤番薯   骑自行车（45分钟）

In [9]:
food.columns = ['food','calorie']
food.head()

food  calorie
0   燕麦片      389
1    酸奶       63
2    水果       60
3  煮鸡胸肉      165
4   炒杂菜       40

In [10]:
sport.columns = ['sport','consume（kcal/m）']
sport.head()

sport  consume（kcal/m）
0     跑步             10.0
1     游泳              9.2
2  健身房训练              8.0
3     瑜伽              3.0
4   骑自行车              8.5

* 检查并去除脏数据

In [11]:
df1[df1['weight(kg)']>150]

Empty DataFrame
Columns: [date, weight(kg), breakfast, lunch, dinner, sport]
Index: []

In [12]:
df1[df1['weight(kg)']<30]

Empty DataFrame
Columns: [date, weight(kg), breakfast, lunch, dinner, sport]
Index: []

In [13]:
food[food['calorie']>1000]

Empty DataFrame
Columns: [food, calorie]
Index: []

In [14]:
food[food['calorie']<=0]

Empty DataFrame
Columns: [food, calorie]
Index: []

In [15]:
sport[sport['consume（kcal/m）']>50]

Empty DataFrame
Columns: [sport, consume（kcal/m）]
Index: []

In [16]:
sport[sport['consume（kcal/m）']<=0]

Empty DataFrame
Columns: [sport, consume（kcal/m）]
Index: []

* 标准化sport

In [17]:
df1['sport']=df1['sport'].apply(lambda x:x.replace('（','('))
df1['sport']=df1['sport'].apply(lambda x:x.replace('）',')'))
df1.head()

date  weight(kg)  breakfast        lunch        dinner        sport
0 2023-06-01       70.20   鸡蛋三明治、牛奶   煮鸡胸肉、青菜、米饭  烤三文鱼、西兰花、烤土豆     跑步(30分钟)
1 2023-06-02       70.10  燕麦片、酸奶、水果  凉拌黄瓜、炒牛肉、米饭    煮虾、炒菜花、烤红薯     游泳(45分钟)
2 2023-06-03       70.30   煎蛋、面包、果汁     番茄炒蛋、炒米粉   煮牛肉、炒蔬菜、烤南瓜     瑜伽(60分钟)
3 2023-06-04       70.00   炒饭、豆浆、水果   煮鸡蛋、炒青菜、米饭  烤鸡胸肉、烤蔬菜、烤豆腐  健身房训练(30分钟)
4 2023-06-05       69.89   面条、煮蛋、果汁  炒鸡胸肉、炒青菜、米饭   煮鱼、凉拌黄瓜、烤番薯   骑自行车(45分钟)

# 数据处理

* 给df1增加每天消耗卡路里的列consume

In [18]:
food_calorie_mid=food.describe().calorie.loc['50%']

In [19]:
def food_to_calorie(x):
    line=x.split('、')
    global sum
    sum=0;
    for i in line:
        if food[food.food==i].empty:
#             print("this is null")
              sum+=food_calorie_mid
        else:
#             print(food[food.food==i]['calorie'].iloc[0])
#             print(type(food[food.food==i]['calorie'].iloc[0]))
            sum+=food[food.food==i]['calorie'].iloc[0]
    return sum

In [20]:
df1['consum_breakfast']=df1['breakfast'].apply(food_to_calorie)
df1.head()

date  weight(kg)  breakfast        lunch        dinner        sport  \
0 2023-06-01       70.20   鸡蛋三明治、牛奶   煮鸡胸肉、青菜、米饭  烤三文鱼、西兰花、烤土豆     跑步(30分钟)   
1 2023-06-02       70.10  燕麦片、酸奶、水果  凉拌黄瓜、炒牛肉、米饭    煮虾、炒菜花、烤红薯     游泳(45分钟)   
2 2023-06-03       70.30   煎蛋、面包、果汁     番茄炒蛋、炒米粉   煮牛肉、炒蔬菜、烤南瓜     瑜伽(60分钟)   
3 2023-06-04       70.00   炒饭、豆浆、水果   煮鸡蛋、炒青菜、米饭  烤鸡胸肉、烤蔬菜、烤豆腐  健身房训练(30分钟)   
4 2023-06-05       69.89   面条、煮蛋、果汁  炒鸡胸肉、炒青菜、米饭   煮鱼、凉拌黄瓜、烤番薯   骑自行车(45分钟)   

   consum_breakfast  
0             209.5  
1             512.0  
2             468.0  
3             411.5  
4             340.0

In [21]:
df1['consum_lunch']=df1['lunch'].apply(food_to_calorie)
df1.head()

date  weight(kg)  breakfast        lunch        dinner        sport  \
0 2023-06-01       70.20   鸡蛋三明治、牛奶   煮鸡胸肉、青菜、米饭  烤三文鱼、西兰花、烤土豆     跑步(30分钟)   
1 2023-06-02       70.10  燕麦片、酸奶、水果  凉拌黄瓜、炒牛肉、米饭    煮虾、炒菜花、烤红薯     游泳(45分钟)   
2 2023-06-03       70.30   煎蛋、面包、果汁     番茄炒蛋、炒米粉   煮牛肉、炒蔬菜、烤南瓜     瑜伽(60分钟)   
3 2023-06-04       70.00   炒饭、豆浆、水果   煮鸡蛋、炒青菜、米饭  烤鸡胸肉、烤蔬菜、烤豆腐  健身房训练(30分钟)   
4 2023-06-05       69.89   面条、煮蛋、果汁  炒鸡胸肉、炒青菜、米饭   煮鱼、凉拌黄瓜、烤番薯   骑自行车(45分钟)   

   consum_breakfast  consum_lunch  
0             209.5         440.5  
1             512.0         393.0  
2             468.0         291.0  
3             411.5         421.0  
4             340.0         421.0

In [22]:
df1['consum_dinner']=df1['dinner'].apply(food_to_calorie)
df1.head()

date  weight(kg)  breakfast        lunch        dinner        sport  \
0 2023-06-01       70.20   鸡蛋三明治、牛奶   煮鸡胸肉、青菜、米饭  烤三文鱼、西兰花、烤土豆     跑步(30分钟)   
1 2023-06-02       70.10  燕麦片、酸奶、水果  凉拌黄瓜、炒牛肉、米饭    煮虾、炒菜花、烤红薯     游泳(45分钟)   
2 2023-06-03       70.30   煎蛋、面包、果汁     番茄炒蛋、炒米粉   煮牛肉、炒蔬菜、烤南瓜     瑜伽(60分钟)   
3 2023-06-04       70.00   炒饭、豆浆、水果   煮鸡蛋、炒青菜、米饭  烤鸡胸肉、烤蔬菜、烤豆腐  健身房训练(30分钟)   
4 2023-06-05       69.89   面条、煮蛋、果汁  炒鸡胸肉、炒青菜、米饭   煮鱼、凉拌黄瓜、烤番薯   骑自行车(45分钟)   

   consum_breakfast  consum_lunch  consum_dinner  
0             209.5         440.5          444.5  
1             512.0         393.0          209.0  
2             468.0         291.0          436.5  
3             411.5         421.0          456.0  
4             340.0         421.0          304.0

In [23]:
df1['consum_day']=df1['consum_breakfast']+df1['consum_lunch']+df1['consum_dinner']
df1.head()

date  weight(kg)  breakfast        lunch        dinner        sport  \
0 2023-06-01       70.20   鸡蛋三明治、牛奶   煮鸡胸肉、青菜、米饭  烤三文鱼、西兰花、烤土豆     跑步(30分钟)   
1 2023-06-02       70.10  燕麦片、酸奶、水果  凉拌黄瓜、炒牛肉、米饭    煮虾、炒菜花、烤红薯     游泳(45分钟)   
2 2023-06-03       70.30   煎蛋、面包、果汁     番茄炒蛋、炒米粉   煮牛肉、炒蔬菜、烤南瓜     瑜伽(60分钟)   
3 2023-06-04       70.00   炒饭、豆浆、水果   煮鸡蛋、炒青菜、米饭  烤鸡胸肉、烤蔬菜、烤豆腐  健身房训练(30分钟)   
4 2023-06-05       69.89   面条、煮蛋、果汁  炒鸡胸肉、炒青菜、米饭   煮鱼、凉拌黄瓜、烤番薯   骑自行车(45分钟)   

   consum_breakfast  consum_lunch  consum_dinner  consum_day  
0             209.5         440.5          444.5      1094.5  
1             512.0         393.0          209.0      1114.0  
2             468.0         291.0          436.5      1195.5  
3             411.5         421.0          456.0      1288.5  
4             340.0         421.0          304.0      1065.0

In [24]:
df1.columns=['date','weight(kg)','breakfast','lunch','dinner','sport','input_breakfast','input_lunch','input_dinner','input_day']
df1.head()

date  weight(kg)  breakfast        lunch        dinner        sport  \
0 2023-06-01       70.20   鸡蛋三明治、牛奶   煮鸡胸肉、青菜、米饭  烤三文鱼、西兰花、烤土豆     跑步(30分钟)   
1 2023-06-02       70.10  燕麦片、酸奶、水果  凉拌黄瓜、炒牛肉、米饭    煮虾、炒菜花、烤红薯     游泳(45分钟)   
2 2023-06-03       70.30   煎蛋、面包、果汁     番茄炒蛋、炒米粉   煮牛肉、炒蔬菜、烤南瓜     瑜伽(60分钟)   
3 2023-06-04       70.00   炒饭、豆浆、水果   煮鸡蛋、炒青菜、米饭  烤鸡胸肉、烤蔬菜、烤豆腐  健身房训练(30分钟)   
4 2023-06-05       69.89   面条、煮蛋、果汁  炒鸡胸肉、炒青菜、米饭   煮鱼、凉拌黄瓜、烤番薯   骑自行车(45分钟)   

   input_breakfast  input_lunch  input_dinner  input_day  
0            209.5        440.5         444.5     1094.5  
1            512.0        393.0         209.0     1114.0  
2            468.0        291.0         436.5     1195.5  
3            411.5        421.0         456.0     1288.5  
4            340.0        421.0         304.0     1065.0

* 为df1增加每天消耗卡路里的列

In [25]:
def sport_to_calorie(x):
    global sum
    line = x.split('(')
    line[1] = line[1][:-3]
    sport_type = line[0]
    time_str = line[1]
    time = int(time_str)
    
    consum_speed=sport[sport.sport=='跑步']['consume（kcal/m）'].iloc[0]
    sum=consum_speed*time
    return sum

In [26]:
df1.head()

date  weight(kg)  breakfast        lunch        dinner        sport  \
0 2023-06-01       70.20   鸡蛋三明治、牛奶   煮鸡胸肉、青菜、米饭  烤三文鱼、西兰花、烤土豆     跑步(30分钟)   
1 2023-06-02       70.10  燕麦片、酸奶、水果  凉拌黄瓜、炒牛肉、米饭    煮虾、炒菜花、烤红薯     游泳(45分钟)   
2 2023-06-03       70.30   煎蛋、面包、果汁     番茄炒蛋、炒米粉   煮牛肉、炒蔬菜、烤南瓜     瑜伽(60分钟)   
3 2023-06-04       70.00   炒饭、豆浆、水果   煮鸡蛋、炒青菜、米饭  烤鸡胸肉、烤蔬菜、烤豆腐  健身房训练(30分钟)   
4 2023-06-05       69.89   面条、煮蛋、果汁  炒鸡胸肉、炒青菜、米饭   煮鱼、凉拌黄瓜、烤番薯   骑自行车(45分钟)   

   input_breakfast  input_lunch  input_dinner  input_day  
0            209.5        440.5         444.5     1094.5  
1            512.0        393.0         209.0     1114.0  
2            468.0        291.0         436.5     1195.5  
3            411.5        421.0         456.0     1288.5  
4            340.0        421.0         304.0     1065.0

In [27]:
df1['consum_day']=df1['sport'].apply(sport_to_calorie)
df1.head()

date  weight(kg)  breakfast        lunch        dinner        sport  \
0 2023-06-01       70.20   鸡蛋三明治、牛奶   煮鸡胸肉、青菜、米饭  烤三文鱼、西兰花、烤土豆     跑步(30分钟)   
1 2023-06-02       70.10  燕麦片、酸奶、水果  凉拌黄瓜、炒牛肉、米饭    煮虾、炒菜花、烤红薯     游泳(45分钟)   
2 2023-06-03       70.30   煎蛋、面包、果汁     番茄炒蛋、炒米粉   煮牛肉、炒蔬菜、烤南瓜     瑜伽(60分钟)   
3 2023-06-04       70.00   炒饭、豆浆、水果   煮鸡蛋、炒青菜、米饭  烤鸡胸肉、烤蔬菜、烤豆腐  健身房训练(30分钟)   
4 2023-06-05       69.89   面条、煮蛋、果汁  炒鸡胸肉、炒青菜、米饭   煮鱼、凉拌黄瓜、烤番薯   骑自行车(45分钟)   

   input_breakfast  input_lunch  input_dinner  input_day  consum_day  
0            209.5        440.5         444.5     1094.5       300.0  
1            512.0        393.0         209.0     1114.0       450.0  
2            468.0        291.0         436.5     1195.5       600.0  
3            411.5        421.0         456.0     1288.5       300.0  
4            340.0        421.0         304.0     1065.0       450.0

* 增加一列表示摄入热量减去运动消耗热量的差值

In [28]:
df1['input_sub_consum_day']=df1['input_day']-df1['consum_day']
df1.head()

date  weight(kg)  breakfast        lunch        dinner        sport  \
0 2023-06-01       70.20   鸡蛋三明治、牛奶   煮鸡胸肉、青菜、米饭  烤三文鱼、西兰花、烤土豆     跑步(30分钟)   
1 2023-06-02       70.10  燕麦片、酸奶、水果  凉拌黄瓜、炒牛肉、米饭    煮虾、炒菜花、烤红薯     游泳(45分钟)   
2 2023-06-03       70.30   煎蛋、面包、果汁     番茄炒蛋、炒米粉   煮牛肉、炒蔬菜、烤南瓜     瑜伽(60分钟)   
3 2023-06-04       70.00   炒饭、豆浆、水果   煮鸡蛋、炒青菜、米饭  烤鸡胸肉、烤蔬菜、烤豆腐  健身房训练(30分钟)   
4 2023-06-05       69.89   面条、煮蛋、果汁  炒鸡胸肉、炒青菜、米饭   煮鱼、凉拌黄瓜、烤番薯   骑自行车(45分钟)   

   input_breakfast  input_lunch  input_dinner  input_day  consum_day  \
0            209.5        440.5         444.5     1094.5       300.0   
1            512.0        393.0         209.0     1114.0       450.0   
2            468.0        291.0         436.5     1195.5       600.0   
3            411.5        421.0         456.0     1288.5       300.0   
4            340.0        421.0         304.0     1065.0       450.0   

   input_sub_consum_day  
0                 794.5  
1                 664.0  
2                 595.5  
3                 988.5  
4                 615.0

In [29]:
df2=df1.loc[:,['date','weight(kg)','input_breakfast','input_lunch','input_dinner','input_day','consum_day','input_sub_consum_day']]
df2.head()

date  weight(kg)  input_breakfast  input_lunch  input_dinner  \
0 2023-06-01       70.20            209.5        440.5         444.5   
1 2023-06-02       70.10            512.0        393.0         209.0   
2 2023-06-03       70.30            468.0        291.0         436.5   
3 2023-06-04       70.00            411.5        421.0         456.0   
4 2023-06-05       69.89            340.0        421.0         304.0   

   input_day  consum_day  input_sub_consum_day  
0     1094.5       300.0                 794.5  
1     1114.0       450.0                 664.0  
2     1195.5       600.0                 595.5  
3     1288.5       300.0                 988.5  
4     1065.0       450.0                 615.0

# 数据展现

* 下图展现了该月的体重
> 体重维持的非常稳定

In [30]:
import pyecharts.options as opts
from pyecharts.charts import Line

x_data = range(1,31)
y_data = df2['weight(kg)']


(
    Line()
    .set_global_opts(
        tooltip_opts=opts.TooltipOpts(is_show=False),
        xaxis_opts=opts.AxisOpts(type_="category"),
        yaxis_opts=opts.AxisOpts(
            type_="value",
            axistick_opts=opts.AxisTickOpts(is_show=True),
            splitline_opts=opts.SplitLineOpts(is_show=True),
            min_ =68
        ),
    )
    .add_xaxis(xaxis_data=x_data)
    .add_yaxis(
        series_name="",
        y_axis=y_data,
        symbol="emptyCircle",
        is_symbol_show=True,
        label_opts=opts.LabelOpts(is_show=False),
    )
    .render_notebook()
)

* 下图展示了每一天摄入卡路里、运动消耗卡路里、热量差的数据
> 每天的卡路里摄入量基本保持在1000到1300之间
> 每天的卡路里运动消耗量基本保持在300到600之间，并且经常以3天为一周期进行300，450，600左右的运动量
> 每天的摄入-运动消耗卡路里基本保持在600到900之间，偶尔有超过900或低于600

In [31]:
import pyecharts.options as opts
from pyecharts.charts import Line

(
    Line()
    .add_xaxis(xaxis_data=range(1,31))
    .add_yaxis(
        series_name="每日摄入卡路里",
#         stack="总量",
        y_axis=df2.input_day,
        label_opts=opts.LabelOpts(is_show=False),
    )
    .add_yaxis(
        series_name="每日运动消耗卡路里",
#         stack="总量",
        y_axis=df2.consum_day,
        label_opts=opts.LabelOpts(is_show=False),
    )
     .add_yaxis(
        series_name="每日摄入-运动消耗卡路里",
#         stack="总量",
        y_axis=df2.input_sub_consum_day,
        label_opts=opts.LabelOpts(is_show=False),
    )
    .set_global_opts(
        title_opts=opts.TitleOpts(title="折线图堆叠"),
        tooltip_opts=opts.TooltipOpts(trigger="axis"),
        yaxis_opts=opts.AxisOpts(
            type_="value",
            axistick_opts=opts.AxisTickOpts(is_show=True),
            splitline_opts=opts.SplitLineOpts(is_show=True),
        ),
        xaxis_opts=opts.AxisOpts(type_="category", boundary_gap=False),
    )
    .render_notebook()
)


* 下图展现了早餐、午餐、晚餐分别占每天总摄入量的比例
> 三餐的摄入量比较平均，午餐为最高，约占37%，早餐和晚餐基本一致

In [32]:
input_breakfast_sum=df2['input_breakfast'].sum()
input_lunch_sum=df2['input_lunch'].sum()
input_dinner_sum=df2['input_dinner'].sum()
input_day_sum=df2['input_day'].sum()

In [33]:
import pyecharts.options as opts
from pyecharts.charts import Pie

a=input_dinner_sum/input_day_sum
b=input_lunch_sum/input_day_sum
c=input_dinner_sum/input_day_sum


x_data = ["早餐", "午餐", "晚餐"]
y_data = [a,b,c]
data_pair = [list(z) for z in zip(x_data, y_data)]
data_pair.sort(key=lambda x: x[1])

(
    Pie(init_opts=opts.InitOpts(bg_color="#2c343c"))
    .add(
        series_name="访问来源",
        data_pair=data_pair,
        rosetype="radius",
        radius="55%",
        center=["50%", "50%"],
        label_opts=opts.LabelOpts(is_show=False, position="center"),
    )
    .set_global_opts(
        title_opts=opts.TitleOpts(
            title="三餐摄入热量所占的比例",
            pos_left="center",
            pos_top="20",
            title_textstyle_opts=opts.TextStyleOpts(color="#fff"),
        ),
        legend_opts=opts.LegendOpts(is_show=False),
    )
    .set_series_opts(
        tooltip_opts=opts.TooltipOpts(
            trigger="item", formatter="{a} <br/>{b}: {c} ({d}%)"
        ),
        label_opts=opts.LabelOpts(color="rgba(255, 255, 255, 0.3)"),
    )
    .render_notebook()
)

# 数据预测

In [34]:
from sklearn.linear_model import LogisticRegression

In [35]:
df2['weight(kg)_str']=df2['weight(kg)'].astype(str)
df2.head()

date  weight(kg)  input_breakfast  input_lunch  input_dinner  \
0 2023-06-01       70.20            209.5        440.5         444.5   
1 2023-06-02       70.10            512.0        393.0         209.0   
2 2023-06-03       70.30            468.0        291.0         436.5   
3 2023-06-04       70.00            411.5        421.0         456.0   
4 2023-06-05       69.89            340.0        421.0         304.0   

   input_day  consum_day  input_sub_consum_day weight(kg)_str  
0     1094.5       300.0                 794.5           70.2  
1     1114.0       450.0                 664.0           70.1  
2     1195.5       600.0                 595.5           70.3  
3     1288.5       300.0                 988.5           70.0  
4     1065.0       450.0                 615.0          69.89

In [36]:
df2['day_num'] = range(1,31)
df2.head()

date  weight(kg)  input_breakfast  input_lunch  input_dinner  \
0 2023-06-01       70.20            209.5        440.5         444.5   
1 2023-06-02       70.10            512.0        393.0         209.0   
2 2023-06-03       70.30            468.0        291.0         436.5   
3 2023-06-04       70.00            411.5        421.0         456.0   
4 2023-06-05       69.89            340.0        421.0         304.0   

   input_day  consum_day  input_sub_consum_day weight(kg)_str  day_num  
0     1094.5       300.0                 794.5           70.2        1  
1     1114.0       450.0                 664.0           70.1        2  
2     1195.5       600.0                 595.5           70.3        3  
3     1288.5       300.0                 988.5           70.0        4  
4     1065.0       450.0                 615.0          69.89        5

## 生成和训练模型

In [37]:
model = LogisticRegression()
model.fit(df2[['day_num']],df2['weight(kg)_str'])

D:\study\lib\site-packages\sklearn\linear_model\_logistic.py:940: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  extra_warning_msg=_LOGISTIC_SOLVER_CONVERGENCE_MSG)


LogisticRegression(C=1.0, class_weight=None, dual=False, fit_intercept=True,
                   intercept_scaling=1, l1_ratio=None, max_iter=100,
                   multi_class='auto', n_jobs=None, penalty='l2',
                   random_state=None, solver='lbfgs', tol=0.0001, verbose=0,
                   warm_start=False)

## 生成未来数据

In [38]:
from datetime import *

In [39]:
df3 = df2[['day_num']].apply(lambda x : x+30)
df3.head()

day_num
0       31
1       32
2       33
3       34
4       35

In [40]:
pred = model.predict(df3[['day_num']])
pred

array(['69.79', '69.79', '69.49', '69.49', '69.49', '69.49', '69.49',
       '69.49', '69.49', '69.49', '69.49', '69.49', '69.49', '69.49',
       '69.49', '69.49', '69.49', '69.49', '69.49', '69.49', '69.49',
       '69.49', '69.49', '69.49', '69.49', '69.49', '69.49', '69.49',
       '69.49', '69.49'], dtype=object)

In [42]:
import pyecharts.options as opts
from pyecharts.charts import Line

x_data = range(1,31)
y_data = pred


(
    Line()
    .set_global_opts(
        tooltip_opts=opts.TooltipOpts(is_show=False),
        xaxis_opts=opts.AxisOpts(type_="category"),
        yaxis_opts=opts.AxisOpts(
            type_="value",
            axistick_opts=opts.AxisTickOpts(is_show=True),
            splitline_opts=opts.SplitLineOpts(is_show=True),
            min_ =68
        ),
    )
    .add_xaxis(xaxis_data=x_data)
    .add_yaxis(
        series_name="",
        y_axis=y_data,
        symbol="emptyCircle",
        is_symbol_show=True,
        label_opts=opts.LabelOpts(is_show=False),
    )
    .render_notebook()
)